datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -682,7 +682,7 @@ impl LogicalPlanBuilder {
             let columns = sort.expr.column_refs();
 
             columns.into_iter().for_each(|c| {
-                if !schema.has_column(c) {
+                if !schema.has_column(c) && !missing_cols.contains(c) {
```

Comment:
can missing cols be a set?

Reply:
I did not try it out, but most likely yes. The change becomes slightly bigger because we currently pass `missing_cols` as a slice to some helper methods.

I assume you are worried about this code now being quadratic? It was not clear if that was a real case in practice that you would order by a large number of columns and none of them are part of the output. But maybe it better to go for the set as it seems safer choice.

Reply:
Done in 557489879e2d4387e820b529a64d28cf5d8476c4